### PR TITLE
Fix heartbeat update mechanism

### PR DESCRIPTION
**Pull request by QuentinBisson**
Fixes https://github.com/giantswarm/giantswarm/issues/25676

## Checklist

I have:

- [x] Described why this change is being introduced
- [x] Separated out refactoring/reformatting in a dedicated PR
- [x] Updated changelog in `CHANGELOG.md`

### CODE REVIEW

**Comment by QuentinBisson** [comment]
When nothing needs to happen:

```json
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"5","message":"checking if heartbeat is already configured","object":"org-lorenzo/h5x7z","resource":"heartbeat","time":"2023-02-01T15:32:21.195709+00:00","version":"1051031831"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"5","message":"heartbeat is configured","object":"org-lorenzo/h5x7z","resource":"heartbeat","time":"2023-02-01T15:32:21.339094+00:00","version":"1051031831"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"5","message":"checking if heartbeat needs to be reconfigured","object":"org-lorenzo/h5x7z","resource":"heartbeat","time":"2023-02-01T15:32:21.339123+00:00","version":"1051031831"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"5","message":"heartbeat is up to date","object":"org-lorenzo/h5x7z","resource":"heartbeat","time":"2023-02-01T15:32:21.339144+00:00","version":"1051031831"}
```

When a new label was added:


```json
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"checking if heartbeat is already configured","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.660632+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"heartbeat is configured","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.805416+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"checking if heartbeat needs to be reconfigured","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.80545+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"heartbeat has changed and needs to be reconfigured","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.805499+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"deleting heartbeat","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.805536+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"deleted heartbeat","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.979243+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"creating heartbeat","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:07.979278+00:00","version":"1051199341"}
{"caller":"github.com/giantswarm/micrologger@v1.0.0/logger.go:66","controller":"prometheus-meta-operator-cluster-api-controller","event":"update","level":"debug","loop":"11","message":"created heartbeat","object":"org-nick/st9q9","resource":"heartbeat","time":"2023-02-01T15:42:08.356677+00:00","version":"1051199341"}
```

File: flag/service/service.go [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package service
 
 import (
-	"github.com/giantswarm/operatorkit/v8/pkg/flag/service/kubernetes"
+	"github.com/giantswarm/operatorkit/v7/pkg/flag/service/kubernetes"
```
Comment (hervenicol) [inline flag/service/service.go:4]: Why the rollback?
Looks like the v8 never even existed, latest release is 7.1.0 :astonished: (https://github.com/giantswarm/operatorkit/blob/master/CHANGELOG.md) - how did it work with v8? :scream_cat:

Comment (QuentinBisson) [reply flag/service/service.go:4]: Well it did not work with 8.0.0 that's the thing. We had a v8 release but it was reverted. I bumped it with renovate last week though

Comment (QuentinBisson) [reply flag/service/service.go:4]: I'm not sure why the renovate pr built then

**Pull request merged**